SalesforceSDK/Core/Settings/IApplicationInformationService.cs

```diff
@@ -33,7 +33,7 @@ public interface IApplicationInformationService
     {
         Task<string> GetApplicationDisplayNameAsync();
 
-        Task<string> GenerateUserAgentHeaderAsync(bool isHybrid);
+        Task<string> GenerateUserAgentHeaderAsync(bool isHybrid, bool isSmartSync);
```

Comment:
For instance on the Android side, the SmartSync code lives in a separate library that depends on the SalesforceSDK library. The base library SalesforceSDK doesn't know/have anything about SmartSync. In the case of the user agent, the SalesforceSDK method simply takes in an extra qualifier string that it appends at the end. For other functionalities, we do the SmartSync specific work in subclasses.